src/dialogs/screenshot/screenshot.vala

```diff
@@ -0,0 +1,1322 @@
+						shoot_screen.begin();
+						return false;
+					});
+					break;
+				case "Window":
+					Timeout.add(200 + (delay * 1000), () => {
+						shoot_window.begin();
+						return false;
+					});
+					break;
+			}
+		}
+
+		async void shoot_window() {
+			bool success = false;
+			string filename_used = "";
+			play_shuttersound(200);
+
+			try {
+				yield client.ScreenshotWindow(include_frame, include_cursor, true, windowstate.tempfile_path, out success, out filename_used);
+			} catch (Error e) {
+				warning("show_window %s, failed to make screenshot\n", e.message);
+				windowstate.statechanged(WindowState.NONE);
+			}
+
+			if (success) {
+				new AfterShotWindow();
+			}
+		}
+
+		private async void shoot_screen() {
+			bool success = false;
+			string filename_used = "";
+
+			play_shuttersound(200);
+
+			try {
+				yield client.Screenshot(include_cursor, true, windowstate.tempfile_path, out success, out filename_used);
+			} catch (Error e) {
+				warning("shoot_screen %s, failed to make screenshot\n", e.message);
+				windowstate.statechanged(WindowState.NONE);
+			}
+
+			if (success) {
+				new AfterShotWindow();
+			}
+		}
+
+		async void shoot_area() {
+			bool success = false;
+			string filename_used = "";
+			int topleftx = this.area[0];
+			int toplefty = this.area[1];
+			int width = this.area[2];
+			int height = this.area[3];
+			// if we just click, forget to drag, set w/h to 1px
+			(height == 0)? height = 1 : height;
+			(width == 0)? width = 1 : width;
+
+			play_shuttersound(0);
+
+			try {
+				yield client.ScreenshotArea(
+					topleftx * scale, toplefty * scale, width * scale, height * scale,
+					include_cursor, true, windowstate.tempfile_path, out success, out filename_used
+				);
+			} catch (Error e) {
+				warning("shoot_area %s, failed to make screenshot\n", e.message);
+				windowstate.statechanged(WindowState.NONE);
+			}
+
+			if (success) {
+				new AfterShotWindow();
+			}
+		}
+
+		private void play_shuttersound(int timeout, string[]? args = null) {
+			if (!emit_capture_sound) return;
+
+			// todo: we should probably not hardcode the soundfile?
+			try {
+				var check = Gst.init_check(ref args);
+				if (!check) {
+					warning("Could not initialise gstreamer");
+					return;
+				}
+			} catch (Error e) {
+				error("Error: %s", e.message);
+			}
+			Gst.Element pipeline;
+
+			try {
+				pipeline = Gst.parse_launch("playbin uri = file:///usr/share/sounds/freedesktop/stereo/screen-capture.oga");
+			} catch (Error e) {
+				error("Error: %s", e.message);
+			}
+
+			Timeout.add(timeout, () => {
+				// fake thread to make sure flash and shutter are in sync
+				pipeline.set_state(State.PLAYING);
+				Gst.Bus bus = pipeline.get_bus();
+				// time out after 2 seconds, in case the output is locked
+				bus.timed_pop_filtered(2000000000, Gst.MessageType.ERROR | Gst.MessageType.EOS);
+				pipeline.set_state(Gst.State.NULL);
+
+				return false;
+			});
+		}
+	}
```

Comment:
You might be able to re-use the code in the Notification server for playing sounds. I seem to recall it being a bit simpler than this.

Reply:
hmm - no real idea how to use CanberraGtk to play that sound file on line 348

Reply:
In the notification server, we have [this](https://github.com/BuddiesOfBudgie/budgie-desktop/blob/7c2ada497de47d52de3259fb138a920ccf913d2e/src/daemon/notifications/dbus.vala#L607-L618) that plays a given sound. You should be able to just do what's in that function, replacing `sound_name` with `"screen-capture"`.

Reply:
Thanks for the pointer.  Tried replacing with this but didn't hear anything.  Thoughts?


    Canberra.Proplist props;
    Canberra.Proplist.create(out props);

    props.sets(Canberra.PROP_CANBERRA_CACHE_CONTROL, "volatile");
    props.sets(Canberra.PROP_EVENT_ID, "screen-capture");

    CanberraGtk.context_get().play_full(0, props);

Come to think of it - I have never ever heard any notification sounds. Didn't realise that was a "thing"

Reply:
Weird, it's working for me when I use the same code you just posted.

Reply:
ok - found the issue - you have alert sounds switched on - BCC - sounds.  If the alert sound is none then nothing is heard.  Very confusing to have an option on the GUI to "play sounds" but dont hear anything because alert sounds is off.  I wonder if its possible to override the muted sound in canberra...?

Reply:
Solved!